src/indra_cogex/client/enrichment/utils.py

```diff
@@ -206,7 +206,10 @@ def collect_genes_with_confidence(
         curie_to_hgnc_ids = defaultdict(dict)
         max_beliefs: Dict[Tuple[str, str, str], float] = {}
         max_ev_counts: Dict[Tuple[str, str, str], int] = {}
-        for result in client.query_tx(query):
+        query_res = client.query_tx(query)
+        if query_res is None:
```

Comment:
if there's an error with execution of the query, then `query_tx` returns None - this commit makes sure that's checked otherwise it's possible to get a more cryptic error when trying to iterate `for result in None`. What kind of error handling should we have here? Should it blow up (with a better error message than just `raise RuntimeError`) or should it potentially also return None so it can be gracefully handled by the caller of `collect_genes_with_confidence`?